### PR TITLE
Fix redundant type casts

### PR DESCRIPTION
**Pull request by fancyweb**
| Q             | A
| ------------- | ---
| Branch?       | 5.3
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

Continuation of https://github.com/symfony/symfony/pull/44274 on 5.3

### CODE REVIEW

File: src/Symfony/Bridge/Doctrine/Form/Type/DoctrineType.php [hunk 1]
```diff
@@ -75,7 +75,7 @@ public static function createChoiceLabel(object $choice): string
      */
     public static function createChoiceName(object $choice, $key, string $value): string
     {
-        return str_replace('-', '_', (string) $value);
+        return str_replace('-', '_', $value);
```
Comment (fancyweb) [inline src/Symfony/Bridge/Doctrine/Form/Type/DoctrineType.php:78]: `string $value` in args

**Review by chalasr — APPROVED** [review]

File: src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler/Configuration.php [hunk 2]
```diff
@@ -107,7 +107,7 @@ private function __construct(array $thresholds = [], $regex = '', $verboseOutput
             if (!isset($this->verboseOutput[$group])) {
                 throw new \InvalidArgumentException(sprintf('Unsupported verbosity group "%s", expected one of "%s".', $group, implode('", "', array_keys($this->verboseOutput))));
             }
-            $this->verboseOutput[$group] = (bool) $status;
+            $this->verboseOutput[$group] = $status;
```
Comment (fancyweb) [inline src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler/Configuration.php:110]: `@param bool[]      $verboseOutput` in the doc + the constructor is private and only called with a `$verboseOutput` in `fromUrlEncodedString()` and `inWeakMode()` and we always set valid booleans values.

File: src/Symfony/Bridge/Twig/Tests/Extension/FormExtensionBootstrap5HorizontalLayoutTest.php [hunk 3]
```diff
@@ -59,7 +59,7 @@ protected function setUp(): void
 
     protected function renderForm(FormView $view, array $vars = []): string
     {
-        return (string) $this->renderer->renderBlock($view, 'form', $vars);
+        return $this->renderer->renderBlock($view, 'form', $vars);
```
Comment (fancyweb) [inline src/Symfony/Bridge/Twig/Tests/Extension/FormExtensionBootstrap5HorizontalLayoutTest.php:62]: `FormRendererInterface::renderBlock()` returns a string

File: src/Symfony/Component/Console/Question/Question.php [hunk 4]
```diff
@@ -105,7 +105,7 @@ public function setHidden(bool $hidden)
             throw new LogicException('A hidden question cannot use the autocompleter.');
         }
 
-        $this->hidden = (bool) $hidden;
+        $this->hidden = $hidden;
```
Comment (fancyweb) [inline src/Symfony/Component/Console/Question/Question.php:108]: `bool $hidden` in args

File: src/Symfony/Component/Console/Question/Question.php [hunk 5]
```diff
@@ -127,7 +127,7 @@ public function isHiddenFallback()
      */
     public function setHiddenFallback(bool $fallback)
     {
-        $this->hiddenFallback = (bool) $fallback;
+        $this->hiddenFallback = $fallback;
```
Comment (fancyweb) [inline src/Symfony/Component/Console/Question/Question.php:130]: `bool $fallback` in args

File: src/Symfony/Component/Console/Question/Question.php [hunk 6]
```diff
@@ -230,11 +230,8 @@ public function getValidator()
      */
     public function setMaxAttempts(?int $attempts)
     {
-        if (null !== $attempts) {
-            $attempts = (int) $attempts;
-            if ($attempts < 1) {
-                throw new InvalidArgumentException('Maximum number of attempts must be a positive value.');
-            }
+        if (null !== $attempts && $attempts < 1) {
```
Comment (fancyweb) [inline src/Symfony/Component/Console/Question/Question.php:233]: `?int $attempts` in args

File: src/Symfony/Component/DependencyInjection/Definition.php [hunk 7]
```diff
@@ -146,7 +146,7 @@ public function setDecoratedService(?string $id, string $renamedId = null, int $
         if (null === $id) {
             $this->decoratedService = null;
         } else {
-            $this->decoratedService = [$id, $renamedId, (int) $priority];
+            $this->decoratedService = [$id, $renamedId, $priority];
```
Comment (fancyweb) [inline src/Symfony/Component/DependencyInjection/Definition.php:149]: `int $priority = 0` in args

File: src/Symfony/Component/DependencyInjection/ParameterBag/ParameterBag.php [hunk 8]
```diff
@@ -109,7 +109,7 @@ public function set(string $name, $value)
      */
     public function has(string $name)
     {
-        return \array_key_exists((string) $name, $this->parameters);
+        return \array_key_exists($name, $this->parameters);
```
Comment (fancyweb) [inline src/Symfony/Component/DependencyInjection/ParameterBag/ParameterBag.php:112]: `string $name` in args

File: src/Symfony/Component/Form/AbstractRendererEngine.php [hunk 9]
```diff
@@ -68,7 +68,7 @@ public function setTheme(FormView $view, $themes, bool $useDefaultThemes = true)
 
         // Do not cast, as casting turns objects into arrays of properties
         $this->themes[$cacheKey] = \is_array($themes) ? $themes : [$themes];
-        $this->useDefaultThemes[$cacheKey] = (bool) $useDefaultThemes;
+        $this->useDefaultThemes[$cacheKey] = $useDefaultThemes;
```
Comment (fancyweb) [inline src/Symfony/Component/Form/AbstractRendererEngine.php:71]: `bool $useDefaultThemes = true` in args

File: src/Symfony/Component/Form/FormFactory.php [hunk 10]
```diff
@@ -66,7 +66,7 @@ public function createNamedBuilder(string $name, string $type = FormType::class,
 
         $type = $this->registry->getType($type);
 
-        $builder = $type->createBuilder($this, (string) $name, $options);
+        $builder = $type->createBuilder($this, $name, $options);
```
Comment (fancyweb) [inline src/Symfony/Component/Form/FormFactory.php:69]: `string $name` in args

File: src/Symfony/Component/Ldap/Adapter/ExtLdap/Adapter.php [hunk 11]
```diff
@@ -72,7 +72,7 @@ public function escape(string $subject, string $ignore = '', int $flags = 0)
         $value = ldap_escape($subject, $ignore, $flags);
 
         // Per RFC 4514, leading/trailing spaces should be encoded in DNs, as well as carriage returns.
-        if ((int) $flags & \LDAP_ESCAPE_DN) {
+        if ($flags & \LDAP_ESCAPE_DN) {
```
Comment (fancyweb) [inline src/Symfony/Component/Ldap/Adapter/ExtLdap/Adapter.php:75]: `int $flags = 0` in args

File: src/Symfony/Component/Translation/Translator.php [hunk 12]
```diff
@@ -455,7 +455,7 @@ protected function computeFallbackLocales(string $locale)
      */
     protected function assertValidLocale(string $locale)
     {
-        if (!preg_match('/^[a-z0-9@_\\.\\-]*$/i', (string) $locale)) {
+        if (!preg_match('/^[a-z0-9@_\\.\\-]*$/i', $locale)) {
```
Comment (fancyweb) [inline src/Symfony/Component/Translation/Translator.php:458]: `string $locale` in args

File: src/Symfony/Component/Validator/Mapping/Loader/AbstractLoader.php [hunk 13]
```diff
@@ -67,7 +67,7 @@ protected function addNamespaceAlias(string $alias, string $namespace)
     protected function newConstraint(string $name, $options = null)
     {
         if (str_contains($name, '\\') && class_exists($name)) {
-            $className = (string) $name;
+            $className = $name;
```
Comment (fancyweb) [inline src/Symfony/Component/Validator/Mapping/Loader/AbstractLoader.php:70]: `string $name` in args

File: src/Symfony/Component/VarDumper/Cloner/Data.php [hunk 14]
```diff
@@ -207,7 +207,7 @@ public function __toString()
     public function withMaxDepth(int $maxDepth)
     {
         $data = clone $this;
-        $data->maxDepth = (int) $maxDepth;
+        $data->maxDepth = $maxDepth;
```
Comment (fancyweb) [inline src/Symfony/Component/VarDumper/Cloner/Data.php:210]: `int $maxDepth` in args

File: src/Symfony/Component/VarDumper/Cloner/Data.php [hunk 15]
```diff
@@ -220,7 +220,7 @@ public function withMaxDepth(int $maxDepth)
     public function withMaxItemsPerDepth(int $maxItemsPerDepth)
     {
         $data = clone $this;
-        $data->maxItemsPerDepth = (int) $maxItemsPerDepth;
+        $data->maxItemsPerDepth = $maxItemsPerDepth;
```
Comment (fancyweb) [inline src/Symfony/Component/VarDumper/Cloner/Data.php:223]: `int $maxItemsPerDepth` in args

**Comment by nicolas-grekas** [comment]
Thank you @fancyweb.

**Pull request merged**